### PR TITLE
Use minimal shell type when encounter CommandInitializationError

### PR DESCRIPTION
**Pull request by lubaihua33**
Some publishers images, such as azhpc-desktop, javlinltd and  vfunctiontechnologiesltd, there might have permission errors when scripts under /etc/profile.d directory are executed at startup of the bash shell for a non-root user. Then calling spawn to run any  Linux commands might raise CommandInitializationError.
The error messages are like this: 
Error while initializing command. The most likely cause is an unsupported shell. Try using a minimal shell type when calling 'spawn' or 'run'. 
(Failed to parse line 'b' /etc/profile.d/clover.sh:line 10:/opt/clover/bin/prepare-hostname.sh: Permission denied'' as integer)                
 
In this PR, except CommandInitializationError, then use minimal shell type.

When waiting to the result, if there is CommandInitializationError when calling spawn, the stdout has that error line before the output of every command. E.g. the stdout of command "uname -vrmo" is like:
 '/etc/profile.d/clover.sh: line 10: /opt/clover/bin/prepare-hostname.sh: Permission denied\r\n0\r\n3.10.0-1160.88.1.el7.x86_6 #1 SMP Tue Mar 7 15:41:52 UTC 2023 x86_64 GNU/Linux'

Other example:
'/etc/profile.d/vglrun.sh: line 3: lspci: command not found\r\nDescription:\tCentOS Linux release 7.9.2009 (Core)'

So in wait_result, filter this error line.

### CODE REVIEW

File: lisa/util/shell.py [hunk 1]
```diff
@@ -290,25 +299,55 @@ def spawn(
...
+                    have_tried_minimal_type = True
+                    matched = _spawn_initialization_error_pattern.search(
+                        str(identifier)
+                    )
+                    if matched:
+                        self.spawn_initialization_error_string = matched.group(
+                            "linux_profile_error"
+                        )
+                else:
+                    raise LisaException(identifier)
```
Comment (squirrelsc) [inline lisa/util/shell.py]: raise identifier is enough.

File: lisa/util/shell.py [hunk 2]
```diff
@@ -290,25 +299,55 @@ def spawn(
...
-        except FunctionTimedOut:
-            raise LisaException(
-                f"The remote node is timeout on execute {command}. "
-                f"It may be caused by paramiko/spur not support the shell of node."
-            )
+        while True:
+            try:
+                if self._inner_shell._spur._shell_type == spur.ssh.ShellTypes.minimal:
+                    # minimal shell type doesn't support store_pid
+                    store_pid = False
```
Comment (squirrelsc) [inline lisa/util/shell.py:308]: Because the pid is not saved, so the process cannot be killed. Any side effect here?

Comment (lubaihua33) [reply lisa/util/shell.py:308]: From my understanding, only when the process timeout, process.kill() is called, then pid is needed. 
So far, in my local testing, I haven't seen any timeout yet. If there is timeout, the worst result might be one process in Linux couldn't be killed, it should not have an impact on the test.

File: lisa/util/shell.py (hunk 1 above)
Comment (lubaihua33) [reply lisa/util/shell.py]: Done

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**